cmd/apprepository-controller/controller_test.go

```diff
@@ -192,8 +192,8 @@ func Test_newCronJob(t *testing.T) {
 				defer func() { userAgentComment = "" }()
 			}
 			result := newCronJob(tt.apprepo)
-			if !reflect.DeepEqual(tt.expected, *result) {
-				t.Errorf("Unexpected result\nExpecting:\n %+v\nReceived:\n %+v", tt.expected, *result)
+			if diff := deep.Equal(tt.expected, *result); diff != nil {
```

Comment:
what's this change for?

Reply:
it's for readability. It failed for me and it just printed both structs. It's' difficult to see the difference but with that `diff` it only prints what's different. For example:

```
    --- FAIL: Test_newSyncJob/my-charts_with_a_custom_pod_template (0.00s)
        /home/andres/projects/src/github.com/kubeapps/kubeapps/cmd/apprepository-controller/controller_test.go:657: [Spec.Template.Spec.Affinity: <nil pointer> != v1.Affinity]
```

Reply:
Can you try to use https://github.com/stretchr/testify since we're using that elsewhere? It also provides better error messages IMO.

Reply:
I may be missing something but I tried `assert.Equal` from testify and it just dumps the structs as well:

```
        /home/andres/projects/src/github.com/kubeapps/kubeapps/cmd/apprepository-controller/controller_test.go:196: 
            	Error Trace:	controller_test.go:196
            	Error:      	Not equal: 
            	            	expected: v1beta1.CronJob{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"apprepo-sync-my-charts", GenerateName:"", Namespace:"kubeapps", SelfLink:"", UID:"", 
...
            	            	actual  : v1beta1.CronJob{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"apprepo-sync-my-charts", GenerateName:"", Namespace:"kubeapps", SelfLink:"", UID:"", 
```

Reply:
Hmm it looks like it's trying to show a diff but for some reason it's not able to? I've seen it show diffs like https://github.com/stretchr/testify/issues/383 in the past.

Reply:
I don't really know, `go-test/deep` seems to do a better job. It doesn't have extra dependencies so I think it's a good addition.

Reply:
Fair enough, just thought it would be worth trying testify